lib/inv_pcg_quda.cpp

```diff
@@ -10,6 +10,9 @@
 #include <invert_quda.h>
 #include <util_quda.h>
 
+#include <madwf_ml.h>
```

Comment:
I'm not sure if we should be including specific reference to the MADWF preconditioner in the PCG file?  Wondering if this can refactored into a more abstract interface so this isn't necessary?

If we end up with specific preconditioner implementation details for every preconditioner we this becomes ugly fast.  Similarly, it's possible we could want to use MADWF with a different solver.  It would be good to decouple the preconditioner and solver.

Reply:
I guess we can have a preconditioner class/interface that handles the creation of the preconditioners?. And yes I also like the idea of decoupling preconditioner and solver.

Reply:
https://github.com/lattice/quda/pull/1061/commits/96926239a741adb4fa5c8bab17dc6714e094af3b